### PR TITLE
Variations: Release Feature

### LINKED ISSUE #3738: [Mobile Payments] Improve Dark Mode Support for Card Reader Settings
The backgrounds on certain table cells is incorrect:

### LINKED ISSUE #3778: Variations: Release Feature
### Tasks

- [x] Remove Feture Flag
- [x] Add Changelog entry

### PR DESCRIPTION
**Pull request by Ecarrion**
Ref: p91TBi-4Ep-p2

closes #3778 

# Why

Since all required tasks for creating variations are merged, now it's time to enable the feature for our 6.2 release.

# How

- Remove the feature flag and all of its usages through our codebase.
- Update release notes

# Testing Steps
Do a smoke test around the feature:

- Create a variation
- Remove a variation
- Create/Edit/Remove/Rename attributes

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -5,6 +5,7 @@
 
 - [*] Fix: Load product inventory settings in read-only mode when the product has a decimal stock quantity. This fixes the products tab not loading due to product decoding errors when third-party plugins enable decimal stock quantities. [https://github.com/woocommerce/woocommerce-ios/pull/3717]
 - [*] Fix: Loading state stuck in Reviews List. [https://github.com/woocommerce/woocommerce-ios/pull/3753]
+- [***] Products: When editing a product, you can now create/delete/update product variations, product attributes and product attribute options. https://github.com/woocommerce/woocommerce-ios/pull/3791
```
Comment (jaclync) [inline RELEASE-NOTES.txt]: nit: I think we can put the higher priority changes higher in the list? it might be easier for the editorial team to see the biggest changes first

File: WooCommerce/Classes/ViewRelated/Products/Edit Product/DefaultProductFormTableViewModel.swift [hunk 2]
```diff
@@ -386,20 +382,13 @@ private extension DefaultProductFormTableViewModel {
                 .map({ String.localizedStringWithFormat(format, $0.name, $0.options.count) })
                 .joined(separator: "\n")
         default:
-            if isAddProductVariationsEnabled {
-                details = ""
-            }
-            else {
-                details = Localization.variationsPlaceholder
-            }
+            details = ""
```
Comment (jaclync) [inline WooCommerce/Classes/ViewRelated/Products/Edit Product/DefaultProductFormTableViewModel.swift:385]: is this change intentional? if so, maybe `Localization.variationsPlaceholder` can be removed?

**Review by jaclync — APPROVED** [review]
Congratulations on closing the variations project! Great work on this complex feature 🎉 PR LGTM 👍 

I think you meant to close https://github.com/woocommerce/woocommerce-ios/issues/3778 instead of Allen's https://github.com/woocommerce/woocommerce-ios/issues/3738? 😄 

Now that the feature is complete, do you think we can:

- Re-enable product type row in product creation form: https://github.com/woocommerce/woocommerce-ios/issues/3050
- Add "Variable product" to product creation - product type selection sheet



I also noticed an edge case during testing:

If there are no variations and I go adding an attribute from the pre-existing attribute list, I can't really select an existing attribute (global or local) unless I update the options. So I was kind of stuck there if I just wanted to pick an existing attribute with the options:


https://user-images.githubusercontent.com/1945542/110076837-1a197180-7dc0-11eb-8e69-a9ccf6b61c1f.mov

**Comment by Ecarrion** [comment]
@jaclync 

> I think you meant to close https://github.com/woocommerce/woocommerce-ios/issues/3778 instead of Allen's https://github.com/woocommerce/woocommerce-ios/issues/3738? 😄 


Ha! Nice catch I always mess up those numbers 🙃 

> - Re-enable product type row in product creation form: https://github.com/woocommerce/woocommerce-ios/issues/3050
> 
> - Add "Variable product" to product creation - product type selection sheet

Wow, I wasn't aware of that, that's for flagging them up!
I'll make sure they are addressed next week! 



> If there are no variations and I go adding an attribute from the pre-existing attribute list, I can't really select an existing attribute (global or local) unless I update the options. So I was kind of stuck there if I just wanted to pick an existing attribute with the options:

Yeah, @ealeksandrov spotted that edge case too and wrote about it in p91TBi-4DP-p2. 

We will address it after getting feedback from the design team!
